pallets/gear/src/lib.rs

```diff
@@ -743,12 +743,15 @@ pub mod pallet {
         ) -> DispatchResultWithPostInfo {
             let who = ensure_signed(origin)?;
 
-            let mut data = Vec::new();
-            // TODO #512
-            code.encode_to(&mut data);
-            salt.encode_to(&mut data);
+            let id: H256 = {
+                let mut data = Vec::new();
```

Comment:
It is better to use `Vec::with_capacity` when we know the length of the resulting vector. This potentially allows avoiding several reallocations.

```suggestion
                let mut data = Vec::with_capacity(code_hash.len() + salt.len());
```